### DISCUSSION TITLE
Source code exposed through URL

### DISCUSSION THREAD

**Discussion by Pittvandewitt** [vercel/community#6143] [Help]
### Summary

I have a simple node.js project written in TypeScript with a single serverless function inside the `api` folder. No framework used and nothing specified in the vercel.json config. Everything seems to work as expected, but there is an issue where I can access all my source files by entering the right URL. Example: [example.vercel.app/api/index.ts](example.vercel.app/api/index.ts). When entering the URL, the file starts downloading. I find this rather alarming. Is this normal behaviour for serverless functions or am I missing something? I have not found anything mentioning this in the docs or on any forum, except for the Directory listing option under settings - advanced, but this only makes things worse.

### Steps to Reproduce

Enter domain/path_to_source_file for a serverless function project and the file starts downloading immediately.

**Comment by coolbii** [comment]
if you want to exploit vercel to serve as a backend server, you have to rewrite all the request under any path to the root 

```
{
  "rewrites": [{ "source": "/(.*)", "destination": "/" }]
}
```
FYI
[https://dev.to/arindam_1729/react-router-not-working-on-vercel-3be6](url)

and make some adjust to output directory

[a comment by an automated account is not shown]
